docs/when-to-update-my-operator.md

```diff
@@ -0,0 +1,26 @@
+# When do I need to update my operator 
```

Comment:
@dmesser it is unclear to me if this was meant to be written for a `cluster administrator` or an `operator author` - could you add context?

Reply:
As a developer - I would look through the `OLM-BOOK` contents and assume that this document tells me when to update an API.